### PR TITLE
adding polsum

### PR DESCRIPTION
**Pull request by kldarek**

### CODE REVIEW

File: datasets/polsum/polsum.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            ),
+        ]
+
+    def _generate_examples(self, filepaths):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        for i, xml_path in enumerate(filepaths):
```
Comment (lhoestq) [inline datasets/polsum/polsum.py:139]:
```suggestion
        for i, xml_path in enumerate(sorted(filepaths)):
```
we need to make the order deterministic here

File: datasets/polsum/polsum.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            {
+                "id": datasets.Value("string"),
+                "date": datasets.Value("string"),
+                "title": datasets.Value("string"),
+                "section": datasets.Value("string"),
+                "authors": datasets.Value("string"),
+                "body": datasets.Value("string"),
+                "summaries": datasets.features.Sequence(
+                    {
+                        "ratio": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/polsum/polsum.py:85]: this must be an integer right ?

You'll need to regenerate the dataset_infos.json file after this change

File: datasets/polsum/README.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+- `body`: a `string` feature.
+- `summaries`: a dictionary feature containing:
+  - `ratio`: a `string` feature.
+  - `type`: a `string` feature.
+  - `author`: a `string` feature.
+  - `body`: a `string` feature.
+  - `spans`: a dictionary feature containing:
+    - `start`: a `int32` feature.
+    - `end`: a `int32` feature.
+    - `span_text`: a `string` feature.
```
Comment (lhoestq) [inline datasets/polsum/README.md:90]: can you add a brief description of each field please after their types ?

File: datasets/polsum/README.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+### Languages
+
+Polish
+
+## Dataset Structure
+
+### Data Instances
+
+[Needs More Information]
```
Comment (lhoestq) [inline datasets/polsum/README.md:72]: can you add one example from the dataset here ?

You can get one using `load_dataset` and taking the first sample `dataset["train"][0]` for example

**Review by lhoestq — COMMENTED** [review]
Awesome thank you for adding this one !

I left a few comments

**Comment by kldarek** [comment]
@lhoestq thanks for the comments! Should be fixed in the latest commit, I assume the CI errors are unrelated.

**Review by lhoestq — APPROVED** [review]
Thanks !

**Pull request merged**